### ISSUE TITLE
DefaultMmapFile memory usage Optimize

### ISSUE THREAD

**Issue by mxsm** [openmessaging/dledger#189]
In DefaultMmapFile  class file have four AtomicInteger properties, as following:

![image](https://user-images.githubusercontent.com/15797831/180629337-f2080548-c819-415c-bc93-d3f165824a91.png)

These attributes should be replaced by AtomicIntegerFieldUpdater。 for Example:

```java
public class AtomicIntegerTest {

    final AtomicInteger startPosition = new AtomicInteger(0);
    final AtomicInteger wrotePosition = new AtomicInteger(0);
    final AtomicInteger committedPosition = new AtomicInteger(0);
    final AtomicInteger flushedPosition = new AtomicInteger(0);

    public static void main(String[] args) throws Exception{
        List<AtomicIntegerTest> list = new LinkedList<>();
        for (int i = 0; i < 1000000; i++) {
            list.add(new AtomicIntegerTest());
        }
        System.out.println("create instances 1000000");
        System.in.read();
    }

}
```

use YourKit tools to analysis, The results are as follows

![image](https://user-images.githubusercontent.com/15797831/180629339-e6192be8-40de-4f2a-a517-34dff233fab0.png)

As you can see **`AtomicInteger`** instances took about 96M memory

```java
public class AtomicIntegerFieldUpdaterTest {

    public static final AtomicIntegerFieldUpdater<AtomicIntegerFieldUpdaterTest> startPosition =  AtomicIntegerFieldUpdater.newUpdater(AtomicIntegerFieldUpdaterTest.class,"startPositionInt");
    public static final AtomicIntegerFieldUpdater<AtomicIntegerFieldUpdaterTest> wrotePosition = AtomicIntegerFieldUpdater.newUpdater(AtomicIntegerFieldUpdaterTest.class,"wrotePositionInt");
    public static final AtomicIntegerFieldUpdater<AtomicIntegerFieldUpdaterTest> committedPosition = AtomicIntegerFieldUpdater.newUpdater(AtomicIntegerFieldUpdaterTest.class,"committedPositionInt");
    public static final AtomicIntegerFieldUpdater<AtomicIntegerFieldUpdaterTest> flushedPosition =AtomicIntegerFieldUpdater.newUpdater(AtomicIntegerFieldUpdaterTest.class,"flushedPositionInt");

    private volatile int startPositionInt = 0;
    private volatile int wrotePositionInt = 0;
    private volatile int committedPositionInt = 0;
    private volatile int flushedPositionInt = 0;

    public static void main(String[] args) throws Exception{
        List<AtomicIntegerFieldUpdaterTest> list = new LinkedList<>();
        for (int i = 0; i < 1000000; i++) {
            list.add(new AtomicIntegerFieldUpdaterTest());
        }
        System.out.println("create instances 1000000");
        System.in.read();
    }

}
```

The same use YourKit tools to analysis, The results are as follows

![image](https://user-images.githubusercontent.com/15797831/180629342-b119c0a5-a9d3-475e-a986-4b83a92d4eac.png)

As you can see **`volatile int + AtomicIntegerFieldUpdater `** instances took about 32M memory

To summarize it, it may pay off to replace Atomic* objects with the corresponding volatile + Atomic*FieldUpdater.

**Comment by mxsm** [comment]
Test DefaultMmapFile, test code:

```java
public class Test {

    public static void main(String[] args) throws Exception{
        List<DefaultMmapFile> list = new LinkedList<>();
        for (int i = 0; i < 100000; i++) {
            list.add(new DefaultMmapFile("C:\\Users\\mxsm\\Desktop\\a\\"+i,1024));
        }
        System.out.println("create instances 100000");
        System.in.read();
    }

}

```

before optimize memory:

![image](https://user-images.githubusercontent.com/15797831/180635168-7c4e9f86-3ebd-4f70-82d6-a8eb8780d536.png)


after optimize memory:

![image](https://user-images.githubusercontent.com/15797831/180635176-cdf7d8f6-c199-4ae7-ba16-15ed29ed1aa2.png)

It reduces about 6M memory
